__1.__ Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз

In [1]:
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def bernoulli(p, n, k):
    return combinations(n,k) * p**k * (1-p)**(n-k)

In [2]:
bernoulli(0.8, 100, 85)

0.048061793700746355

__2.__ Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [3]:
import numpy as np

def poisson(m, n, p):
    l = p * n
    return l**m/factorial(m) * np.exp(-l)

In [4]:
p_zero = poisson(0, 5000, 0.0004)
p_zero

0.1353352832366127

In [5]:
p_two = poisson(2, 5000, 0.0004)
p_two

0.2706705664732254

__3.__ Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [6]:
bernoulli(0.5,144,70)

0.06281178035144776

__4.__ В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

In [7]:
white_all = combinations(7,2)/combinations(10,2) * combinations(9,2)/combinations(11,2)
white_all

0.3054545454545455

In [8]:
good_two = combinations(7,2)*combinations(2,2) + \
combinations(7,1)*combinations(3,1)*combinations(9,1)*combinations(2,1) + \
combinations(9,2)*combinations(3,2)
good_all = combinations(10,2) * combinations(11,2)
white_two = good_two/good_all
white_two

0.20484848484848484

In [16]:
p_white_from_first_two = combinations(7,2)/combinations(10,2) * combinations(2,2)/combinations(11,2)
p_white_from_first_and_second_one = (combinations(7,1)/combinations(10,1)*combinations(3,1)/combinations(9,1))+\
(combinations(9,1)/combinations(11,1)*combinations(2,1)/combinations(10,1))
p_white_from_second_two = combinations(3,2)/combinations(10,2) * combinations(9,2)/combinations(11,2)
white_two = p_white_from_first_two + p_white_from_first_and_second_one + p_white_from_second_two
white_two

0.44909090909090904

In [10]:
#Третий вариант решения:
#Возможны следующие сочетания вытянутых мячей: (ww)(..);(w.)(w.);(.w)(w.);(.w)(.w);(w.)(.w);(..)(ww)
#из этих сочетаний вывожу следующие вероятности:
7/10*6/9*2/11*1/10 + 7/10*3/9*9/11*2/10 + 3/10*7/9*9/11*2/10 + 3/10*7/9*2/11*9/10 + 7/10*3/9*2/11*9/10 + 3/10*2/9*9/11*8/10

0.20484848484848484

Вывод:
- задание сделано тремя методами, два из которых схожи по исполнению, но ни один из методов не вызывает у меня доверия. так как вероятность вытащить все четыре белых мяча больше вероятности вытащить хотя бы 2, хотя в виду количества белых мячей в обеих корзинах вариант с вероятностью в 0.2048(48) впролне возможен;
- до сих пор не понимаю принцип и место применения терминов "Совместные и несовместные события" и "Вероятность зависимых и независимых событий".

In [11]:
not_white_all = combinations(3,2)/combinations(10,2)*combinations(2,2)/combinations(11,2)
1 - not_white_all

0.9987878787878788